File: src/main/java/io/github/toberocat/core/utility/events/faction/FactionDeleteEvent.java
package io.github.toberocat.core.utility.events.faction;

import io.github.toberocat.core.utility.factions.Faction;
import org.bukkit.entity.Player;

public class FactionDeleteEvent extends FactionEventCancelledable {

    public FactionDeleteEvent(Faction faction) {
        super(faction);
    }
}

File: improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/api/events/FactionDeleteEvent.kt
package io.github.toberocat.improvedfactions.api.events

import io.github.toberocat.improvedfactions.database.storage.FactionSnapshot
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

/**
 * Fired when a faction gets deleted.
 */
class FactionDeleteEvent(val faction: FactionSnapshot) : Event(), Cancellable {
    private var isCancelled = false

    override fun getHandlers(): HandlerList = handlerList
    
    override fun isCancelled(): Boolean = isCancelled

    override fun setCancelled(cancel: Boolean) {
        isCancelled = cancel
    }

    companion object {
        @JvmStatic
        val handlerList = HandlerList()
    }
}
